writers/local.go

```diff
@@ -0,0 +1,14 @@
+package writers
```

Comment:
so generally between packages you want to create a public interface that exposes a function (or two) and then you create structs for each of these that implement that interface (the structs can / should be private). Then based on your config you can create one of those structs, and then put that interface into your config. It'll clean up your logic a bit in root.go as well.

https://gitlab.morningconsult.com/mci/dev-ops/hodor/-/merge_requests/38/diffs

the cache implementation there is a decent example.

Reply:
So do you mean that in the entrypoint you would have something like?:
```
doc := MakeDoc(config)
doc.upload()
doc.print()
doc.markdown
```